### DISCUSSION TITLE
Autoupdate always asks for a password (Target SDK 10.13+,  Xcode14.3)

### DISCUSSION THREAD

**Discussion by pawel-gener8** [sparkle-project/Sparkle#2390] [Q&A]
## Summary

We use the sparkle v1.20 in a project but we had to move to the MacOS SDK 13.3. This forced us to change the SDK target minimum version for Sparkle from 10.9 to 10.13. However after this change the autoupdate app always asks for an admin password (in order to remove the downloaded file from quarantine). Is this something known? Is there a cure?

## Possible Fix

?

## Version

1.20

**Comment by zorgiepoo** [comment]
I doubt that change is the cause of this. Did you enable sandboxing in your app? Sandboxing is only supported in Sparkle 2 and there is a guide for it.

**Reply by pawel-gener8** [reply]
Nope. The only change was switch to SDK13.3 (Xcode 14.3) that forced `MACOSX_DEPLOYMENT_TARGET` bump.

**Reply by pawel-gener8** [reply]
I've just double checked and the app isn't sandboxed (Activity Monitor confirms that). I'm however not entirely true about the fact this was "the only change". That is true only in the context of sparkle but the app was actively developed of course. I'll move to the older version and compile it with `MACOSX_DEPLOYMENT_TARGET` bumped and Xcode 14.3 and see if it manifests the same behaviour or that is something triggered by a change in the app itself. I'll get back to you once I know it. Thanks

**Reply by pawel-gener8** [reply]
Good call. It's something in the product that introduced that - I built an older version that updates silently using Xcode 14.3 with the development target bumped and it still installs silently. I'll dig more on my side, then,
I saw that in the 2.x version in case of the error when trying to remove the quarantine attrib Sparkle does not even bother to elevate as this error does not seem fatal - it does not cancel the update. Would it be reasonable to follow that in 1.2?

**Reply by zorgiepoo** [reply]
Quarantine removal is not a fatal error in 1.x either. Sparkle asks for authorization when it does not otherwise have sufficient write access to replace your app.

Does your app update succeed if you do authorize or does it still fail? If it still fails, find the error log message in your app or Autoupdate. You can also cross-verify what the behavior in 2.x is for debugging this.

Another check: can you say, mv the app in Terminal without requiring sudo / root?

**Reply by pawel-gener8** [reply]
It does succeed even if not granted admin rights so it's only about removing this quarantine bit.

**Reply by zorgiepoo** [reply]
Which file is it trying to remove the quarantine bit of? The download for the update or one of Sparkles helpers? Paste the error log.

Also there is a chance this does not occur in Sparkle 2 anyway as the authorization model there changed. This is something you could quickly test out using sparkle 2’s SUUpdater adapter temporarily.

But to be clear I haven’t seen a user hit this kind of issue before. What OS are you running too? You can also try to bisect the change that somehow “introduced” this in your project.

**Reply by pawel-gener8** [reply]
It's the `Autoupdate.app`. The error log simply is:
`Sparkle: Authorization required to remove quarantine file:///var/folders/t2/z7x7gqsd3svbq_s399dhfpkc0000gn/T/TemporaryItems/NSIRD_Autoupdate_U78ohA/X.app`
and then
`Failed to remove file quarantine on X.app`
It's on macos 13.4 and I'm sure it does not happen in 2.x because the code is no longer there - 2.x just does not ask for elevation in this case just moves on.
BUT
I was able to nail it down to one resource being RO for everyone. Making it RW makes this elevation dialog no to show.
Thanks for the help. Let's close this.
